accounts/abi/bind/base.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package bind
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// SignerFn is a signer function callback when a contract requires a method to
+// sign the transaction before submission.
+type SignerFn func(common.Address, *types.Transaction) (*types.Transaction, error)
+
+// CallOpts is the collection of options to fine tune a contract call request.
+type CallOpts struct {
+	Pending bool // Whether to operate on the pending state or the last known one
+}
+
+// TransactOpts is the collection of authorization data required to create a
+// valid Ethereum transaction.
+type TransactOpts struct {
+	Account common.Address // Ethereum account to send the transaction from
```

Comment:
I'll sleep on it ;)

On Wed, Mar 23, 2016 at 7:35 PM, bas-vk <notifications@github.com> wrote:

> In accounts/abi/bind/base.go
> <https://github.com/ethereum/go-ethereum/pull/2357#discussion_r57202850>:
>
> > +	"github.com/ethereum/go-ethereum/crypto"
> > +)
> > +
> > +// SignerFn is a signer function callback when a contract requires a method to
> > +// sign the transaction before submission.
> > +type SignerFn func(common.Address, *types.Transaction) (*types.Transaction, error)
> > +
> > +// CallOpts is the collection of options to fine tune a contract call request.
> > +type CallOpts struct {
> > +	Pending bool // Whether to operate on the pending state or the last known one
> > +}
> > +
> > +// TransactOpts is the collection of authorization data required to create a
> > +// valid Ethereum transaction.
> > +type TransactOpts struct {
> > +	Account common.Address // Ethereum account to send the transaction from
>
> I still prefer the use of from ;-)
> Guess its a matter of personal preference.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ethereum/go-ethereum/pull/2357/files/67f1d565173e3a5a69d3f6fb2279ba22a839bc68#r57202850>
>